### PR TITLE
error fix broken path and permission error

### PR DESCRIPTION
**Pull request by shing1Sks**
## Issue

1.Due to an extra leading '/' or '\' in the dirname variabe a broken path such as path: 'C:\C:\Users\SHREYASH%20KUMAR%20SINGH\Desktop\hackfest\electricitymaps-contrib\web\public' like this is generated
2.An EPERM error

## Description

Added an extra check to check if an extra '/' or '\' exists and removing it if it does for the issue 1

Wrapped the directory check code in try catch block to better handle the issues related and it handles the EPERM error

### Double check

- [*] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [*] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/scripts/generateSitemap.js [hunk 1]
```diff
@@ -5,14 +5,28 @@ import path from 'node:path';
...
 import zonesConfig from '../config/zones.json' assert { type: 'json' };
 
-const __dirname = path.dirname(new URL(import.meta.url).pathname);
+let dirname = path.dirname(new URL(import.meta.url).pathname);
+
+// Fix the paths for Windows/Linux consistency
+// Extra / or \\ in begenning gives a broken path
+if (dirname.startsWith('/') || dirname.startsWith('\\')) {
+  dirname = dirname.substring(1);
+}
```
[the remark this answers is not in the document]
Comment (shing1Sks) [reply web/scripts/generateSitemap.js:14]: okay my bad 

its just in my local system C:\C:\Users\SHREYASH%20KUMAR%20SINGH\Desktop\hackfest\electricitymaps-contrib\web\public this path was created with the extra slash and resolved when removed

i will retry my setup and configurations

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply web/scripts/generateSitemap.js:14]: Yeah it's not great if it's not working on windows, we really need it to work on all platforms (Windows, MacOS, and Linux) so it's great that you are looking into it!

[the remark this answers is not in the document]
Comment (shing1Sks) [reply web/scripts/generateSitemap.js]: now i have used fileURLToPath it should handle windows and linux paths evenly accross the platform 
image
this is the path now in my system 
could you please check now if it works !

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply web/scripts/generateSitemap.js]: It works for me as well! I'll approve the PR and get this merged.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**